### PR TITLE
Pagination for long list of pages

### PR DESCRIPTION
**Pull request by AZeghers**
Implementation of a more user-friendly pagination for long lists of pages
All specifications can be found in the related documentation page
@ostoc Feel free to give your feedback

### CODE REVIEW

**Comment by ostoc** [comment]
Looks pretty good, well done

File: packages/kotti-pagination/src/components/FlexiblePaginator.vue [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+				'page-item': true,
+				[className]: this.currentPage === page,
+			}
+		},
+		humanReadablePageNumber(page) {
+			return page + 1
+		},
+	},
+	computed: {
+		pixelMargin() {
```
Comment (FlorianWendelborn) [inline packages/kotti-pagination/src/components/FlexiblePaginator.vue]: I think you can just put it in a box and center it, making calculations nearly unnecessary.

File: packages/kotti-pagination/src/components/FlexiblePaginator.vue [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+			const digitWidth = this.maximumPage.toString().length - 1
+			const numberWidth = BASE_NUMBER_WIDTH + digitWidth * PIXEL_MULTIPLIER
+			const baseElementsWidth =
+				(2 * this.adjacentAmount + MAX_NUMBER_OF_ELEMENTS) * numberWidth
+			const adjacentPagesOffset = this.adjacentAmount * ADJACENT_MULTIPLIER
+			return baseElementsWidth + BASE_OFFSET * 2 + adjacentPagesOffset
+		},
+		containerStyle() {
+			if (this.fixedWidth) return 'width: ' + this.pixelMargin + 'px'
```
Comment (FlorianWendelborn) [inline packages/kotti-pagination/src/components/FlexiblePaginator.vue]:
```suggestion
			return this.fixedWidth ? { width: `${this.pixelMargin}px` } :  {}
```

File: packages/kotti-pagination/src/components/FlexiblePaginator.vue [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+			return ''
```
Comment (FlorianWendelborn) [inline packages/kotti-pagination/src/components/FlexiblePaginator.vue]:
```suggestion
```

**Comment by AZeghers** [comment]
I am merging this, as Dominik Lindenberger requested this for PR to go through.
I will pick up the calculations again in the future to comply with @FlorianWendelborn 's requests.

**Comment by FlorianWendelborn** [comment]
> in the future

(X)

**Pull request merged**